src/state/mod.rs

```diff
@@ -128,17 +132,20 @@ impl State {
 
     pub fn input(&mut self, event: &WindowEvent) -> bool {
         match event {
-            &WindowEvent::KeyboardInput {
+            WindowEvent::KeyboardInput {
                 // Challenge tutorial Pipeline
                 input:
                     KeyboardInput {
-                        state: ElementState::Pressed,
+                        state,
                         virtual_keycode: Some(VirtualKeyCode::Space),
                         ..
                     },
                 ..
-            } => false,
-            &WindowEvent::CursorMoved {
+            } => {
+                self.use_color = *state == ElementState::Released;
```

Comment:
why we don't match directly this value in the destructuring of `KeyboardInput` above (like what was done before with `ElementState::Pressed`)?
```rust
                    KeyboardInput {
                        state: ElementState::Pressed,
                        virtual_keycode: Some(VirtualKeyCode::Space),
                        ..
                    },
```

Reply:
`use_color` is now renamed to `bool_event`, which is a property of `State`. It's a hardcode boolean, which allows to "toggle" some configurations of the render pipeline's output. Its functionality can be found in `update()` method of struct `State`
```rust
        match self.bool_event {
            true => {
                self.vertex_buffer = Buffer::new(
                    &self.device,
                    Vertex::VERTICES_PENTAGON,
                    wgpu::BufferUsages::VERTEX,
                    "Vertex Buffer",
                );
                self.index_buffer = Some(Buffer::new(
                    &self.device,
                    Vertex::INDICES,
                    wgpu::BufferUsages::INDEX,
                    "Index Buffer",
                ));
                self.num_indices = Vertex::INDICES.len() as u32;
                self.texture_bind_group = Texture::init_texture_bind_group(
                    &self.device,
                    &self.default_texture,
                    &self.texture_bind_group_layout,
                    "diffuse_texture_bind_group",
                );
            }
            false => {
                self.vertex_buffer = Buffer::new(
                    &self.device,
                    &new_vertex,
                    wgpu::BufferUsages::VERTEX,
                    "Challenge Vertex Buffer",
                );
                self.index_buffer = Some(Buffer::new(
                    &self.device,
                    &new_indices,
                    wgpu::BufferUsages::INDEX,
                    "Index Buffer",
                ));
                self.num_indices = new_indices.len() as u32;
                self.texture_bind_group = Texture::init_texture_bind_group(
                    &self.device,
                    &self.cartoon_texture,
                    &self.texture_bind_group_layout,
                    "cartoon_texture_bind_group",
                );
            }
        };

```